fix: enable hot reload for the products dir

skoob13 opened the pull request:
## Problem

The products dir is up and awesome, but the hot reloading doesn't work.

## Changes

- Add the `products` path to the esbuild config.
- React to manifest changes because it only updates for js/ts module files by default.

## Does this work well for both Cloud and self-hosted?

Yes

## How did you test this code?

Manually

skoob13 committed:
fix: enable hot reload in the products dir

mariusandra reviewed (approved):
Seems like a flakey test, I've seen the same before. However, yes, thanks for this! 

One improvement suggestion though: adding `common/` into the list might also make sense

Twixes commented on frontend/utils.mjs:
```diff
@@ -369,6 +369,7 @@ export async function buildOrWatch(config) {
                 [
                     path.resolve(absWorkingDir, 'src'),
                     path.resolve(absWorkingDir, '../ee/frontend'),
+                    path.resolve(absWorkingDir, '../products'),
```
Let's only include changes to frontend files, not backend

skoob13 committed:
fix: watch common and frontend-related files only

skoob13 commented:
@mariusandra I've added `common` to the list. It will work only for modules directly imported in the frontend tree.

@Twixes I've updated the watch config, but it might not be necessary, as the rebuild only works for the modules imported in the project, so Python will trigger a chokidar event, but it won't trigger an update.

skoob13 committed:
Merge branch 'master' of github.com:PostHog/posthog into fix/products-dir-hot-reload

skoob13 committed:
fix: sneaky console.log
